### PR TITLE
Add AMD/ROCm configs and workshop for ORNL Frontier

### PR DESCRIPTION
**Pull request by PranavViswanath**
Every cluster in the almanac is NVIDIA. Frontier is AMD MI250X, and every config in `nvidia_configs/` hardcodes a CUDA wheel index. Modal has no AMD GPUs, so the Modal workshop cannot derive the ROCm equivalents. This ports the loop to a rented MI300X and ships what it produced.

## amd_configs/

Five ROCm environment specs, each probe-verified on an MI300X (real energy and forces on an 8-atom Cu crystal) and gated for Frontier's gfx90a:

| model | checkpoint | energy | probe | gfx90a gate |
|---|---|---|---|---|
| mace | mace-mp-0-medium | -32.645 eV | OK | PASS |
| chgnet | chgnet-default | -32.649 eV | OK | PASS |
| mattersim | mattersim-v1-0-0-5m | -32.757 eV | OK | PASS |
| orb | orb-v2 | -32.710 eV | OK | PASS |
| uma | uma-s-1p1 | -29.979 eV | OK | PASS |

Four independently-trained networks landing within 0.11 eV of each other on the same crystal is the sanity check that ROCm numerics are not quietly wrong.

**Every `setup()` is byte-identical to its `nvidia_configs/` twin.** The entire port lives in the PEP 723 dependency block. PyTorch's ROCm build exposes AMD GPUs through the `cuda` API, so no model code changes and `_agent/probe.py` is untouched.

## amd_workshop/

The ROCm counterpart of `modal_app.py`: one uv venv per config (the same path `rootstock install` uses on HPC) instead of Modal images, ssh instead of `modal run`. Plus two gates the probe cannot replace.

**`check_isa.py` (Frontier gate).** GPU code is compiled per-architecture. A build carrying only gfx942 (MI300X) imports fine, probes green, and then dies on Frontier with `HSA_STATUS_ERROR_INVALID_ISA`. The probe structurally cannot catch this, since it passes on the build box. Verified: official ROCm torch wheels are fat binaries and do carry gfx90a (23 of 23 fat libraries), so pure-wheel models are Frontier-safe for free.

**`check_numerics.py` (correctness gate).** A build that succeeds is not a build that is correct. `torch-sparse` compiles on ROCm, imports, runs without error, and returns wrong numbers:

```
dense reference  : [[2, 2, 2], [7, 7, 7]]
torch_sparse GPU : [[2, 2, 2], [2, 2, 2]]     <-- no error raised
```

Its `spmm` kernel is built around a 32-lane NVIDIA warp; AMD wavefronts are 64 lanes, so its shuffle-based row reduction drops contributions. The probe passes and the physics is corrupt. This gate compares every source-built GPU op against a CPU reference.

## _agent/failure_modes.md

New ROCm section. The load-bearing entries:

- **`pytorch-triton-rocm` is published only on the ROCm index, never on PyPI.** ROCm torch depends on it, so a naive CUDA-to-ROCm config cannot even resolve. This failure cannot occur on NVIDIA (the CUDA equivalent is on PyPI), so it is invisible from the NVIDIA side. Fix: declare it as a direct dependency AND map it in `[tool.uv.sources]`. A sources entry alone does nothing, since uv only routes packages that appear in `dependencies`.

- **The obvious fix for that is a trap.** Relaxing index isolation (dropping `explicit = true`) so the resolver finds it does work, while silently resolving `torchmetrics` to 1.0.3 instead of 1.9.0, because the ROCm index mirrors a stale slice of PyPI and uv takes each package from the first index that carries it. The result is a working environment quietly running different library code, which is the worst case for reproducible science. Keep `explicit = true` and declare the missing package. When a package is missing, add the package; never widen the search.

- **Same failure, one layer down.** `--index-strategy unsafe-best-match` with an unpinned torch installs a CUDA build (2.13.0+cu130) onto an AMD box, and `torch.cuda.is_available()` returns False. It is safe only against a pinned lockfile, which is how `workshop.py` uses it.

- Plus: lockfile installs need the config's indexes passed through, and a ROCm torch env is roughly 17 GB (the wheel alone is 4.2 GB, about double CUDA's), which has real filesystem implications on a cluster.

## What this does not prove

Nothing has executed on an MI250X. This validates dependency resolution and the API layer on ROCm, and confirms Frontier's machine code is present in the shipped binaries. It does not prove that code runs there. Someone with a Frontier allocation should `rootstock install` these and confirm.

Also untested: the `molecule` and `slab_co` probe systems (only `crystal` was run), and Rootstock's own i-PI worker path, which the workshop deliberately bypasses just as `modal_app.py` does.

## Follow-up, not in this PR

The PyG-extension tier (fairchem OC20 models, TensorNet, TorchMD-Net) needs `torch-scatter` / `torch-sparse` / `torch-cluster`, for which no ROCm wheels exist anywhere. Progress so far, for a later PR:

- Source-building works. The recipe is a ROCm toolchain matching the torch wheel's ROCm version, installing from git rather than PyPI, and `PYTORCH_ROCM_ARCH="gfx90a;gfx942"`. The resulting `.so` carries both architectures, so one artifact serves Frontier and the MI300X. You do not need an MI250X to compile for one.
- `torch-scatter` and `torch-cluster` build dual-arch and are numerically correct.
- `torch-sparse`'s `spmm` is not (see above). Notably fairchem never imports `torch_sparse`, so that tier may be less blocked than it appears.

## Open questions for maintainers

- Directory layout: `amd_configs/` mirrors `nvidia_configs/`, but you may prefer `configs/{amd,nvidia}/`.
- `check_isa.py` is arguably not workshop-specific. It is a Frontier-readiness check that could live alongside `rootstock install` on the cluster itself.

### CODE REVIEW

**Review by OwenPriceSkelly — APPROVED** [review]
LGTM
